MontyTRC89/Tomb-Editor
# Rearrange node arguments

Provides a way to rearrange node arguments by fetching variable names from node function's signature.

To test:

1. Try to make a test project with very complex node setups on a master branch
2. switch to that rearrange branch
3. open the project and make sure that all nodes have correct arguments, and nothing was broken or wrong values appeared
4.  save the project, go to any node catalog with nodes that you placed for test and shuffle arguments 
5. also, make sure you also shuffle arguments in function signature.

I will provide an example now. Let's imagine you have opened Inventory.lua node catalog 
there is a function:

```
-- !Name "Add item to inventory"
-- !Section "Inventory"
-- !Description "Add item to the inventory."
-- !Description "A count of 0 will add the default pickup amount of that item."
-- !Arguments "NewLine, 80, WadSlots, [ _ITEM ], Object to add to Lara's inventory."
-- !Arguments "20, Numerical, [ 0 | 1000 ], Amount of items to add (0 to give default amount)."

LevelFuncs.Engine.Node.AddInventoryItem = function(item, count)
    TEN.Inventory.GiveItem(item, count)
end
try to shuffle args in this way:
-- !Name "Add item to inventory"
-- !Section "Inventory"
-- !Description "Add item to the inventory."
-- !Description "A count of 0 will add the default pickup amount of that item."
-- !Arguments "NewLine, 20, Numerical, [ 0 | 1000 ], Amount of items to add (0 to give default amount)."
-- !Arguments "80, WadSlots, [ _ITEM ], Object to add to Lara's inventory."

LevelFuncs.Engine.Node.AddInventoryItem = function(count, item)
    TEN.Inventory.GiveItem(item, count)
end
```
 
then open your re-saved project again and make sure everything is okay
and please do it for various types of nodes.
